contrib/spring/src/test/java/io/opencensus/contrib/spring/instrument/web/TraceWebAsyncClientAutoConfigurationTest.java

```diff
@@ -0,0 +1,227 @@
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.opencensus.trace.Span.Kind.CLIENT;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+import io.opencensus.common.Scope;
+import io.opencensus.contrib.http.util.HttpTraceAttributeConstants;
+import io.opencensus.testing.export.TestHandler;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceParams;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter;
+import io.opencensus.trace.samplers.Samplers;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(
+    classes = {TraceWebAsyncClientAutoConfigurationTest.TestConfiguration.class},
+    properties = "opencensus.spring.enabled=true",
+    webEnvironment = RANDOM_PORT)
+@SuppressWarnings("deprecation")
+public class TraceWebAsyncClientAutoConfigurationTest {
+  @Autowired org.springframework.web.client.AsyncRestTemplate asyncRestTemplate;
+
+  @Autowired Environment environment;
+
+  Tracer tracer;
+
+  private TestHandler handler;
+
+  @Before
+  public void setup() {
+    handler = new TestHandler();
+
+    SpanExporter exporter = Tracing.getExportComponent().getSpanExporter();
+    exporter.registerHandler("testing", handler);
+
+    TraceParams params =
+        Tracing.getTraceConfig()
+            .getActiveTraceParams()
+            .toBuilder()
+            .setSampler(Samplers.alwaysSample())
+            .build();
+    Tracing.getTraceConfig().updateActiveTraceParams(params);
+  }
+
+  @Test(timeout = 10000)
+  @Order(1)
+  // Need to suppress warnings for MustBeClosed because Java-6 does not support try-with-resources.
+  @SuppressWarnings("MustBeClosedChecker")
+  public void should_close_span_upon_success_callback()
+      throws ExecutionException, InterruptedException {
+    this.tracer = Tracing.getTracer();
```

Comment:
Curious - why 'this.tracer' rather than 'tracer'?

Reply:
copy/paste. Removed.